appIcons.js

```diff
@@ -426,6 +446,15 @@ function activateAllWindows(app, settings) {
     }
 }
 
+/**
+ * This closes all windows of the app.
+ */
+function closeAllWindows(app, settings) {
+    let windows = getInterestingWindows(app, settings);
+    for (let i = 0; i < windows.length; i++)
+        windows[i].delete(global.get_current_time());
+}
+
```

Comment:
Now that we have this function, it could be used  also for the quit menu (line 549), and so the `_closeWindowInstance` method (line 528) could be removed.